commit/chainfee/processor.go

```diff
@@ -3,41 +3,126 @@ package chainfee
 import (
 	"context"
 	"fmt"
+	"math/big"
+	"time"
 
 	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"golang.org/x/exp/maps"
+
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	readerpkg "github.com/smartcontractkit/chainlink-ccip/pkg/reader"
 
 	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
 )
 
-type Processor struct {
+type processor struct {
+	destChain  cciptypes.ChainSelector
+	lggr       logger.Logger
+	homeChain  reader.HomeChain
+	ccipReader readerpkg.CCIPReader
+	cfg        pluginconfig.CommitOffchainConfig
+	fRoleDON   int
 }
 
-func NewProcessor() *Processor {
-	return &Processor{}
+// nolint: revive
+func NewProcessor(
+	lggr logger.Logger,
+	destChain cciptypes.ChainSelector,
+	homeChain reader.HomeChain,
+	ccipReader readerpkg.CCIPReader,
+	offChainConfig pluginconfig.CommitOffchainConfig,
+	fRoleDON int,
+) *processor {
+	return &processor{
+		lggr:       lggr,
+		destChain:  destChain,
+		homeChain:  homeChain,
+		ccipReader: ccipReader,
+		fRoleDON:   fRoleDON,
+		cfg:        offChainConfig,
+	}
 }
 
-func (w *Processor) Query(ctx context.Context, prevOutcome Outcome) (Query, error) {
+func (p *processor) Query(ctx context.Context, prevOutcome Outcome) (Query, error) {
 	return Query{}, nil
 }
 
-func (w *Processor) Observation(
+func (p *processor) Observation(
 	ctx context.Context,
 	prevOutcome Outcome,
 	query Query,
 ) (Observation, error) {
-	return Observation{}, nil
+	// Get the fee components for all available chains that we can read from
+	feeComponents := p.ccipReader.GetAvailableChainsFeeComponents(ctx)
+	// Get the native token prices for all available chains that we can read from
+	nativeTokenPrices := p.ccipReader.GetWrappedNativeTokenPriceUSD(ctx, maps.Keys(feeComponents))
+	// Get the latest chain fee price updates for the source chains
+	chainFeePriceUpdates := p.ccipReader.GetChainFeePriceUpdate(ctx, maps.Keys(feeComponents))
+
+	return Observation{
+		FChain:            p.ObserveFChain(),
+		FeeComponents:     feeComponents,
+		NativeTokenPrices: nativeTokenPrices,
+		ChainFeeUpdates:   chainFeePriceUpdates,
+		Timestamp:         time.Now().UTC(),
+	}, nil
 }
 
-func (w *Processor) Outcome(
+func (p *processor) Outcome(
 	prevOutcome Outcome,
 	query Query,
 	aos []plugincommon.AttributedObservation[Observation],
 ) (Outcome, error) {
-	return Outcome{}, nil
+
+	consensusObs, err := p.getConsensusObservation(aos)
+	if err != nil {
+		return Outcome{}, err
+	}
+	// No need to update yet
+	if len(consensusObs.FeeComponents) == 0 {
+		return Outcome{}, nil
```

Comment:
Worth adding a log here so we can track behavior of this component and aid in debugging?

```suggestion
		p.lggr.Debug("no consensus on fee components, nothing to update", "consensusObs", consensusObs)
		return Outcome{}, nil
```